packages/main/src/utils/Strings.js

```diff
@@ -0,0 +1,86 @@
+/**
+	* Escape a regular expression text so that it can be used in a regular expression.
+	* @param {string} text The raw text
+	* @returns The regular expression text, with special characters having been escaped.
+	*/
+export function escapeRegex(text) {
+	return text.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+/**
+	* Replace all instances of a particular find text.˝
+	* @param {string} text The text to find/replace on
+	* @param {string} find The string to find
+	* @param {string|function} replace The replace string or function
+	* @param {boolean=} caseInsensitive true if the replace is case insensitive, false by default
+	* @return {string} The string with all matches replaced.
+	*/
+export function replaceAll(text, find, replace, caseInsensitive) {
+	return text.replace(new RegExp(escapeRegex(find), `${caseInsensitive ? "i" : ""}g`), replace);
+}
+
+// HTML needs to escape all attribute special characters, plus a few extras
+const dangerousReplacements = {
+	'&lt;': /</g,
+	'&amp;': /&/g,
+	'&quot;': /"/g,
+	'&gt;': />/g,
+	'&#39;': /'/g
```

Comment:
NOTE: This code actually compresses better than my original version.

Reply:
**[thought (non-blocking)](https://conventionalcomments.org/#labels):** Considering what I said in https://github.com/SAP/ui5-webcomponents/pull/4246#discussion_r742513411, we may want change this to `&apos;` instead of `&#39;`.